3scaleAdapter/pkg/threescale/metrics/prometheus.go

```diff
@@ -79,6 +89,19 @@ func (r *Reporter) ObserveBackendLatency(backendURL string, serviceID string, ob
 	}
 }
 
+// ReportBackendStatus reports a hit to 3scale backend and reports status code of the result
+func (r *Reporter) ReportBackendStatus(code int, backendURL string, serviceID string) error {
```

Comment:
why not make a method that receives the Prometheus' `CounterVec` object and +1's the url, service_id, code? IOW, you could just pass in `backendStatusCodes` and perhaps this could be reusable for system?

Reply:
do you mean an internal method or do you mean to expose the Prometheus collectors outside of the package?

Reply:
Also I think these will diverge actually because backend needs an additional label to note the function called, ie `authorise or report`

Reply:
is there a way in which we could add the additional label and then pass it to a method like this one, which just adds the common labels and `Inc()`s it?

Reply:
I'm thinking about avoiding the `Backend` infix in all these methods so that we don't repeat essentially the same for system ones.